pkg/provider/kubernetes/client.go

```diff
@@ -300,15 +312,42 @@ func convertMap(in map[string][]byte) map[string]string {
 	return out
 }
 
-func (c *Client) createSecret(ctx context.Context, value []byte, typed v1.SecretType, remoteRef esv1beta1.PushSecretRemoteRef) error {
+func (c *Client) createSecret(ctx context.Context, secret *v1.Secret, typed v1.SecretType, remoteRef esv1beta1.PushSecretData) error {
+	data := make(map[string][]byte)
+
+	if remoteRef.GetProperty() != "" {
+		// set a specific remote key
+		if remoteRef.GetSecretKey() == "" {
+			// we push the whole secret if secret key was not set
+			values := make(map[string]string)
+			for k, v := range secret.Data {
+				values[k] = string(v)
+			}
+			// marshal
+			value, err := json.Marshal(values)
```

Comment:
@moolen Sooo... When defining a property they probably want the whole thing pushed as is, right?

Therefor I thought if we marshal it up that would be okay? I have to convert it to string though, otherwise the marshal will marshal bytes which will end up in numbers.

I think I know we do this in vault or something. Do you have a better way of doing this?

Reply:
Awesome! Thanks.

I will get to the other providers soon. 🤣

Reply:
It was right in the same package OMG 🤦 😆